azurestack/resource_arm_route_table.go

```diff
@@ -0,0 +1,272 @@
+	tags := d.Get("tags").(map[string]interface{})
+
+	routes, err := expandRouteTableRoutes(d)
+	if err != nil {
+		return fmt.Errorf("Error Expanding list of Route Table Routes: %+v", err)
+	}
+
+	routeSet := network.RouteTable{
+		Name:     &name,
+		Location: &location,
+		RouteTablePropertiesFormat: &network.RouteTablePropertiesFormat{
+			Routes: &routes,
+		},
+		Tags: *expandTags(tags),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, name, routeSet)
+	if err != nil {
+		return fmt.Errorf("Error Creating/Updating Route Table %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for completion of Route Table %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, name, "")
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Route Table %q (resource group %q) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmRouteTableRead(d, meta)
+}
+
+func resourceArmRouteTableRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).routeTablesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["routeTables"]
+
+	resp, err := client.Get(ctx, resGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Route Table %q: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureStackNormalizeLocation(*location))
+	}
+
+	if props := resp.RouteTablePropertiesFormat; props != nil {
+		if err := d.Set("route", flattenRouteTableRoutes(props.Routes)); err != nil {
+			return err
+		}
+
+		if err := d.Set("subnets", flattenRouteTableSubnets(props.Subnets)); err != nil {
+			return err
+		}
+	}
+
+	flattenAndSetTags(d, &resp.Tags)
+
+	return nil
+}
+
+func resourceArmRouteTableDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).routeTablesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["routeTables"]
+
+	future, err := client.Delete(ctx, resGroup, name)
+	if err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error deleting Route Table %q (Resource Group %q): %+v", name, resGroup, err)
+		}
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for deletion of Route Table %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func expandRouteTableRoutes(d *schema.ResourceData) ([]network.Route, error) {
```

Comment:
This function never returns an error, so error can be removed from the return values.